# **Waze User Churn Project - Part 3: Hypothesis Testing**

In this notebook, the data team will perform hypothesis testing to determine whether there is a significant difference in the average number of drives between users based on their device type.

This notebook aims to answer the question: 

> **"Is there a significant difference in the average number of drives between iPhone users and Android users?"**

### **Imports and data loading**




In [ ]:
# Import any relevant packages or libraries
import pandas as pd
from scipy import stats

In [2]:
# Load dataset into dataframe
df = pd.read_csv('waze_dataset.csv')

### **Data exploration**

In this part of the notebook, label encoding was performed, where categorical text data was converted into unique numerical integers.

The following devices were assigned to the following values:
* **`iPhone - 1`**
* **`Android - 2`**

In [3]:
# 1. Create `map_dictionary`
map_dictionary = {'Android': 2, 'iPhone': 1}

# 2. Create new `device_type` column
df['device_type'] = df['device'].map(map_dictionary)

# 3. Map the new column to the dictionary
df['device_type']

0        2
1        1
2        2
3        1
4        2
        ..
14994    1
14995    2
14996    1
14997    1
14998    1
Name: device_type, Length: 14999, dtype: int64

Descriptive statistics is performed so that it will be easier to compare the average amount of drives by device type.

In [ ]:
# 1 - iPhone
# 2 - Android
df.groupby('device_type')['drives'].mean()

device_type
1    67.859078
2    66.231838
Name: drives, dtype: float64

Based on the averages shown, it appears that drivers who use an iPhone device to interact with the application have a higher number of drives on average. However, this difference might arise from random sampling, rather than being a true difference in the number of drives. To assess whether the difference is statistically significant, the team conducts hypothesis testing.


### **Hypothesis testing**

A two-sample t-test was performed.

**The hypotheses are as follows:**
* Null Hypothesis: There is no difference in the average number of rides between iPhone and Android users.
* Alternative Hypothesis: There is a difference in the average number of rides between iPhone and Android users.

The significance level was set at 5%.

In [ ]:
# 1. Isolate the data for iPhone users.
iphone = df[df['device'] == 'iPhone']

# 2. Isolate the data for Android users.
android = df[df['device'] == 'Android']

# 3. Perform the t-test
# set equal_var to False since we do not have a strong reason to assume that both groups have the same variance
stats.ttest_ind(a = iphone['drives'],
                b = android['drives'],
                equal_var = False)

Ttest_indResult(statistic=1.4635232068852353, pvalue=0.1433519726802059)

Since the p-value **`(0.14)`** is **greater than** the significance level **`(0.05)`**, we fail to reject the null hypothesis. Therefore, there is no significant difference in the average number of drives between Android and iPhone users.

### **Key Takeaways**

* Based on the hypothesis test, there is no significant difference in the average number of drives between Android and iPhone users.
* Since device type does not appear to affect driving activity, further investigation is needed to identify the factors that contribute to the variation in users' driving activity. Additional hypothesis tests can also be conducted to better understand user behavior and its relationship with churn.